web/src/app/converters/submission-data-converter.ts

```diff
@@ -97,9 +91,10 @@ function taskDataPbToModel(pb: Pb.ITaskData[], job: Job): SubmissionData {
         coordinatesPbToModel(captureLocationResult.coordinates!)
       );
     else if (takePhotoResult) value = takePhotoResult.photoPath;
-    else throw new Error('Error converting to Submission: invalid task data');
+    else if (!skipped)
+      throw new Error('Error converting to Submission: invalid task data');
 
-    submissionData[task.id] = new Result(value!);
+    submissionData[taskId!] = new Result(value!);
```

Comment:
What is `value` when `skipped` is `true`? If it's `null` or `undefined`, won't `value!` throw an error? If that's not the case, is there a way to make this code more self-explanatory?